src/Block/AdminPreviewBlockService.php

```diff
@@ -0,0 +1,135 @@
+ */
+final class AdminPreviewBlockService extends AbstractBlockService
+{
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    public function __construct(Environment $twig, Pool $pool)
+    {
+        parent::__construct($twig);
+
+        $this->pool = $pool;
+    }
+
+    public function execute(BlockContextInterface $blockContext, ?Response $response = null): Response
+    {
+        $admin = $this->getAdmin($blockContext->getSetting('code'));
+        $this->handleFilters($admin, $blockContext);
+
+        foreach ($blockContext->getSetting('remove_list_fields') as $listField) {
+            $admin->getList()->remove($listField);
+        }
+
+        $datagrid = $admin->getDatagrid();
+
+        return $this->renderPrivateResponse($blockContext->getTemplate(), [
+            'block' => $blockContext->getBlock(),
+            'settings' => $blockContext->getSettings(),
+            'admin' => $admin,
+            'datagrid' => $datagrid,
+        ], $response);
+    }
+
+    public function getName()
+    {
+        return 'Admin preview';
+    }
+
+    public function configureSettings(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'text' => 'Preview',
+            'filters' => [],
+            'icon' => false,
+            'limit' => 10,
+            'code' => false,
+            'template' => '@SonataAdmin/Block/block_admin_preview.html.twig',
+            'remove_list_fields' => [ListMapper::NAME_ACTIONS],
+        ]);
+    }
+
+    /**
+     * @throws \InvalidArgumentException if the provided admin code is invalid
+     */
+    private function getAdmin(string $code): AdminInterface
+    {
+        try {
+            $admin = $this->pool->getAdminByAdminCode($code);
+        } catch (ServiceNotFoundException $e) {
+            throw new \InvalidArgumentException('Unable to find the Admin instance.', $e->getCode(), $e);
+        }
+
+        if (!$admin instanceof AdminInterface) {
+            throw new \InvalidArgumentException('The requested service is not an Admin instance.');
+        }
+
+        $admin->checkAccess('list');
+
+        return $admin;
+    }
+
+    /**
+     * Maps the block filters to standard admin filters.
+     */
+    private function handleFilters(AdminInterface $admin, BlockContextInterface $blockContext): void
+    {
+        $filters = $blockContext->getSetting('filters');
+
+        if ($sortBy = $filters['_sort_by'] ?? null) {
+            $sortFilters = ['_sort_by' => $sortBy];
+            if ($sortOrder = $filters['_sort_order'] ?? null) {
+                $sortFilters['_sort_order'] = $sortOrder;
+                unset($filters['_sort_order']);
+            }
+            // Setting a request to the admin is a workaround since the admin only
+            // can handle the "_sort_by" parameter from the query string.
+            $request = new Request(['filter' => $sortFilters]);
+            $request->setSession(new Session());
```

Comment:
Do we really need a session here?

Reply:
Yes, I'm afraid: https://github.com/sonata-project/SonataAdminBundle/blob/f9fbd32d4f14d3e40d6d9ea90f17adb4b7510bf7/src/Admin/AbstractAdmin.php#L2695-L2702

Reply:
Does the request has a hasSession or something? We could add a check to the code

Reply:
Sure, I was thinking the same.
Let me check and I'll update the PR if possible.

Reply:
Here you are: #7080.